src/test/js/runtests.js

```diff
@@ -95,6 +95,12 @@ function runTests() {
 
 var Drone = require("src/main/js/modules/drone/index.js");
 
+function testDroneWithoutArgumentUsesSelfAsPlayer() {
+  global.self = {name: 'joe', location:{ x:10, y:15, z:20 }};
+  var drone = new Drone();
+  assertEqual([10, 15, 23], [drone.x, drone.y, drone.z]);
```

Comment:
Need to make sure `global.self` does not leak into other tests (reset in a `finally` block?)